request.go

```diff
@@ -1106,3 +1130,35 @@ func (req *ExecuteRequest) Context(ctx context.Context) *ExecuteRequest {
 	req.ctx = ctx
 	return req
 }
+
+// IdRequest informs the server about supported protocol
+// version and features.
+type IdRequest struct {
+	baseRequest
+	idInfo
+}
+
+// newIdRequest returns a new IdRequest.
+func newIdRequest(version ProtocolVersion, features []Feature) *IdRequest {
```

Comment:
Request type is public, but a user can not to create an object in the right way (`NewIdRequest`).

Reply:
Changed